Integrations/AWS-EC2/AWS-EC2.py

```diff
@@ -0,0 +1,2679 @@
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    obj = vars(client._client_config)
+    kwargs = {}
+    data = []
+
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('ownerIds') is not None:
+        kwargs.update({'OwnerIds': parse_resource_ids(args.get('ownerIds'))})
+    if args.get('snapshotIds') is not None:
+        kwargs.update({'SnapshotIds': parse_resource_ids(args.get('snapshotIds'))})
+    if args.get('restorableByUserIds') is not None:
+        kwargs.update({'RestorableByUserIds': parse_resource_ids(args.get('restorableByUserIds'))})
+
+    response = client.describe_snapshots(**kwargs)
+
+    for i, snapshot in enumerate(response['Snapshots']):
+        data.append({
+            'Description': snapshot['Description'],
+            'Encrypted': snapshot['Encrypted'],
+            'OwnerId': snapshot['OwnerId'],
+            'Progress': snapshot['Progress'],
+            'SnapshotId': snapshot['SnapshotId'],
+            'StartTime': datetime.strftime(snapshot['StartTime'], '%Y-%m-%dT%H:%M:%SZ'),
+            'State': snapshot['State'],
+            'VolumeId': snapshot['VolumeId'],
+            'VolumeSize': snapshot['VolumeSize'],
+            'Region': obj['_user_provided_options']['region_name'],
+        })
+        if 'Tags' in snapshot:
+            for tag in snapshot['Tags']:
+                data[i].update({
+                    tag['Key']: tag['Value']
+                })
+
+    output = json.dumps(response['Snapshots'], cls=DatetimeEncoder)
+    raw = json.loads(output)
+    raw[0].update({'Region': obj['_user_provided_options']['region_name']})
+    ec = {'AWS.EC2.Snapshots(val.SnapshotId === obj.SnapshotId)': raw}
+    human_readable = tableToMarkdown('AWS EC2 Snapshots', data)
+    return_outputs(human_readable, ec)
+
+
+def describe_volumes_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+
+    obj = vars(client._client_config)
+    kwargs = {}
+    data = []
+
+    if args.get('filters') is not None:
+        kwargs.update({'Filters': parse_filter_field(args.get('filters'))})
+    if args.get('volumeIds') is not None:
+        kwargs.update({'VolumeIds': parse_resource_ids(args.get('volumeIds'))})
+
+    response = client.describe_volumes(**kwargs)
+
+    for i, volume in enumerate(response['Volumes']):
+        data.append({
+            'AvailabilityZone': volume['AvailabilityZone'],
+            'Encrypted': volume['Encrypted'],
+            'State': volume['State'],
+            'VolumeId': volume['VolumeId'],
+            'VolumeType': volume['VolumeType'],
+            'CreateTime': datetime.strftime(volume['CreateTime'], '%Y-%m-%dT%H:%M:%SZ'),
```

Comment:
this can throw value error. better calculate it before the append and in a try block.